wrappers/core18.go

```diff
@@ -0,0 +1,160 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2018 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package wrappers
+
+import (
+	"bytes"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"regexp"
+
+	"github.com/snapcore/snapd/dirs"
+	"github.com/snapcore/snapd/osutil"
+	"github.com/snapcore/snapd/release"
+	"github.com/snapcore/snapd/snap"
+	"github.com/snapcore/snapd/systemd"
+)
+
+func writeSnapdToolingMountUnit(sysd systemd.Systemd, prefix string) error {
+	content := []byte(fmt.Sprintf(`[Unit]
+Description=Make the snapd snap tooling available for the system
+Before=snapd.service
+
+[Mount]
+What=%s/usr/lib/snapd
+Where=/usr/lib/snapd
+Type=none
+Options=bind
+
+[Install]
+RequiredBy=snapd.service
+`, prefix))
+	unit := "usr-lib-snapd.mount"
+	_, _, err := osutil.EnsureDirState(dirs.SnapServicesDir,
```

Comment:
I think you could use `EnsureFileState` as it's for a signle file and not the whole directory.
I also think we could and should integrate these changes into `(*systemd.systemd).WriteMountUnitFile` (in particular the use of `EnsureFileState` to avoid rewriting the same file will make embedded happier).

Lastly, given `EnsureFileState` returns a flag value when the files were the same, this info should be used to avoid the `DaemonReload` if no  change was made. This'll make some people happy.

----
alternatively, return just the content of the file and use that to pass it into the call to `EnsureDirStateGlobs`.